src/sas/qtgui/Plotting/Slicers/BoxSlicer.py

```diff
@@ -250,101 +346,158 @@ def setParams(self, params):
         :param params: a dictionary containing name of slicer parameters and
             values the user assigned to the slicer.
         """
-        self.x = float(numpy.fabs(params["x_max"]))
-        self.y = float(numpy.fabs(params["y_max"]))
+        self.half_width = params["half_width"]
+        self.half_height = params["half_height"]
         self.nbins = params["nbins"]
         self.fold = params["fold"]
-
-        self.horizontal_lines.update(x=self.x, y=self.y)
-        self.vertical_lines.update(x=self.x, y=self.y)
+        self.center_x = params["center_x"]
+        self.center_y = params["center_y"]
+
+        # save current state of the ROI in case the change leaves no data in
+        # the ROI and thus a disallowed move. Also set the has_move flags to
+        # true in case we have to restore this saved state.
+        self.save(ev=None)
+        self.center.has_move = True
+        self.horizontal_lines.has_move = True
+        self.vertical_lines.has_move = True
+        # Now update the ROI based on the change
+        self.center.update(center_x=self.center_x, center_y=self.center_y)
+        self.horizontal_lines.update(center=self.center,
+                                     half_width=self.half_width, half_height=self.half_height)
+        self.vertical_lines.update(center=self.center,
+                                   half_width=self.half_width, half_height=self.half_height)
+        # Compute and plot the 1D average based on these parameters
         self._post_data()
+        # Now move is over so turn off flags
+        self.center.has_move = False
+        self.horizontal_lines.has_move = False
+        self.vertical_lines.has_move = False
         self.draw()
 
+    def validate(self, param_name, param_value):
+        """
+        Validate input from user.
+        Values get checked at apply time.
+        * nbins cannot be zero or samller
+        * The full ROI should stay within the data. thus center_x and center_y
+          are restricted such that the center +/- width (or height) cannot be
+          greate or smaller than data max/min.
+        * The width/height should not be set so small as to leave no data in
+          the ROI. Here we only make sure that the width/height is not zero
+          as done when dragging the vertical or horizontal lines. We let the
+          call to _post_data capture the ValueError of no points in ROI
+          raised by manipulations.py, log the message and negate the entry
+          at that point.
+        """
+        isValid = True
+
+        if param_name =='half_width':
+            # Can't be negative for sure. Also, it should not be so small that
+            # there remains no points to average in the ROI. We leave this
+            # second check to manipulations.py
+            if param_value <= 0:
+                logging.warning("The box width is too small. Please adjust.")
```

Comment:
Actually not true in this particular case since that is in the Validate function which only applies to text entries.  It also is never true I don't think for the "no points in ROI" which is captured in a try except block. However it is the case with all other illegal mouse drags.  I did find that annoying but not sure how to fix. I will need to think a bit more about this. I'm not sure that "unselecting" will not be more annoying to the user. Suspending the error till `isValid == True` maybe? but have to make sure I capture it everywhere it can be called....

Reply:
OK ... This looks like it might work. See the commit message for explanation of `BaseInteractor.py` changes.  What do you think?